### ISSUE TITLE
Qubes Builder fails at handling branches with new files

### ISSUE THREAD

**Issue by marmarta** [QubesOS/qubes-issues#9423]
### Brief summary

Qubes Builder after fetch leaves files in the repository.

### Steps to reproduce
1. have a branch with some new files in a repo, everything is clean and committed
2. run `./qb -c repo package fetch` on this repo
3. get switched back to main branch in this repo 

### Expected behavior
Have a clean repo.

### Actual behavior

The new files from the old branch are lying around uncommited in the repo.

**Comment by fepitre** [comment]
Is this similar to Git behavior when having new files in a branch?

**Comment by marmarek** [comment]
Normally when you switch between branches (`git checkout <branch>` or `git switch <branch>`), git removes files that are not present in the other branch. Here, for some reason it doesn't. I guess it's related to some weird way we switch branches

**Comment by marmarta** [comment]
Yeah, as @marmarek said: when I switch branches, the files appear and vanish according to what a given branch contains, but not when I do `fetch` with qubes builder v2.

**Comment by RonaldRonnie** [comment]
is it that the untracked files have to be only in the new such that the main is clean unless commited?

**Comment by RonaldRonnie** [comment]
Assign me please to this issue
